crates/polars-core/src/datatypes/any_value.rs

```diff
@@ -415,6 +415,7 @@ impl<'a> AnyValue<'a> {
                     NumCast::from(0)
                 }
             },
+            String(v) => NumCast::from((*v).parse::<i64>().ok()?),
```

Comment:
Can you try something like this? `(*v).parse::<i128>().ok_or_else(|v| v.parse::<f64>())`

Reply:
@ritchie46 `parse` results a `Result` not an `Option` so we can't use `ok_or_else`. Is `map_err` the way to go here, or is there a better "do this if the result is an error"? I can't find an "unwrap_or_else" function that doesn't require an input to the closure (and that input would just be an `Err(x)` where `x` is probably an error message).

```rust
String(v) => NumCast::from(
    (*v).parse::<i128>().map_err(|_| (*v).parse::<f64>()).ok()?
),
```

Reply:
@ritchie46 I ended up going with this:

```rust
String(v) => match (*v).parse::<i128>() {
    Ok(val) => NumCast::from(val),
    Err(_) => NumCast::from((*v).parse::<f64>().ok()?),
},
```

Not sure if there's a more elegant way of doing that. I don't think I can chain them into the same `NumCast` call because the first parse returns a `Result<i128,_>` and the other a `Result<f64,_>`.

Reply:
This should be possible via an iterator adaptor. Will take a look later.

Reply:
I guess even just an `if let` would be better.